packages/interactivity/src/directives.js

```diff
@@ -205,21 +205,21 @@ const cssStringToObject = ( val ) => {
  * @param {string} type 'window' or 'document'
  * @return {void}
  */
-const getGlobalEventDirective =
-	( type ) =>
-	( { directives, evaluate } ) => {
+const getGlobalEventDirective = ( type ) => {
+	return ( { directives, evaluate } ) => {
 		directives[ `on-${ type }` ]
 			.filter( ( { suffix } ) => suffix !== 'default' )
 			.forEach( ( entry ) => {
+				const event = entry.suffix.split( '--', 1 )[ 0 ];
 				useInit( () => {
-					const cb = ( event ) => evaluate( entry, event );
+					const cb = () => evaluate( entry, event );
```

Comment:
This breaks the event handlers, it's replacing the event passed to the event handler with the name of the event in an outer scope:

```suggestion
					const cb = ( event ) => evaluate( entry, event );
```

Reply:
Done in https://github.com/WordPress/gutenberg/pull/61009/commits/a8189567d1001b02b32f704d6efe2ca2ea52d66d